docs/transports.md

```diff
@@ -0,0 +1,362 @@
+
+Create a route handler using Slim's built-in factories and container:
+
+```php
+use Psr\Container\ContainerInterface;
+use Slim\Factory\AppFactory;
+use Slim\Psr7\Factory\ResponseFactory;
+use Slim\Psr7\Factory\StreamFactory;
+use Mcp\Server;
+use Mcp\Server\Transport\StreamableHttpTransport;
+
+$app = AppFactory::create();
+$container = $app->getContainer();
+
+$container->set('mcpServer', function (ContainerInterface $container) {
+    return Server::builder()
+        ->setServerInfo('My MCP Server', '1.0.0')
+        ->setDiscovery(__DIR__, ['.'])
+        ->build();
+});
+
+$app->any('/mcp', function ($request, $response) {
+    $mcpServer = $this->get('mcpServer');
+    
+    $responseFactory = new ResponseFactory();
+    $streamFactory = new StreamFactory();
+    
+    $transport = new StreamableHttpTransport($request, $responseFactory, $streamFactory);
+    $mcpServer->connect($transport);
+    
+    return $transport->listen();
+});
+```
+
+### HTTP Method Handling
+
+The transport handles all HTTP methods automatically:
+
+- **POST**: Send MCP requests
+- **GET**: Not implemented (returns 405)
+- **DELETE**: End session
+- **OPTIONS**: CORS preflight
+
+You should route **all methods** to your MCP endpoint, not just POST.
+
+### Session Management
+
+HTTP transport requires persistent sessions since PHP doesn't maintain state between requests. Unlike STDIO transport
+where in-memory sessions work fine, HTTP transport needs a persistent session store:
+
+```php
+use Mcp\Server\Session\FileSessionStore;
+
+// ✅ Good for HTTP
+$server = Server::builder()
+    ->setSession(new FileSessionStore(__DIR__ . '/sessions'))
+    ->build();
+
+// ❌ Not recommended for HTTP (sessions lost between requests)
+$server = Server::builder()
+    ->setSession(new InMemorySessionStore())
+    ->build();
+```
+
+### Recommended Route
+
+It's recommended to mount the MCP endpoint at `/mcp`, but this is not enforced:
+
+```php
+// Recommended
+Route::any('/mcp', [McpController::class, 'handle']);
+
+// Also valid
+Route::any('/', [McpController::class, 'handle']);
+Route::any('/api/mcp', [McpController::class, 'handle']);
+```
+
+### Testing HTTP Transport
+
+Use the MCP Inspector to test HTTP servers:
+
+```bash
+# Start your PHP server
+php -S localhost:8000 server.php
+
+# Connect with MCP Inspector
+npx @modelcontextprotocol/inspector http://localhost:8000
+```
+
+## Choosing a Transport
+
+### Use STDIO When:
+
+- Building command-line tools
+- Integrating with MCP clients (Claude Desktop, etc.)
+- Simple request/response patterns
+- Process-based isolation is desired
+- Single session per process
+
+### Use HTTP When:
+
+- Building web services
+- Need browser compatibility
+- Integration with web frameworks
+- Multiple concurrent sessions
+- Web-based MCP clients
+- RESTful API patterns
+
+The key difference is that STDIO is process-based (one session per process) while HTTP is request-based (multiple sessions via headers).
```

Comment:
not sure i agree here - for me the key difference is connectivity. are you able to connect via STDIO since it's local? no, go HTTP.